samcli/local/docker/lambda_container.py

```diff
@@ -133,11 +137,60 @@ def __init__(
             entrypoint=_entrypoint if _entrypoint else entry,
             env_vars=env_vars,
             container_opts=additional_options,
-            additional_volumes=additional_volumes,
+            additional_volumes={**additional_volumes, **layer_volume_mounts},
             container_host=container_host,
             container_host_interface=container_host_interface,
         )
 
+    @staticmethod
+    def _get_layer_folder_mounts(layers: List[LayerVersion]) -> Dict[str, Dict[str, str]]:
+        """
+        Searches the code uri of the Layer to resolve any root level symlinks before
+        the container is created
+
+        Paramters
+        ---------
+        layers: List[LayerVersion]
+            A list of layers to check for any symlinks
+
+        Returns
+        -------
+        Dict[str, Dict[str, str]]
+            A dictonary representing the resolved file or directory and the bound path
+            on the container
+        """
+        layer_mappings: Dict[str, Dict[str, str]] = {}
+
+        for layer in layers:
+            # layer.compatible_runtimes can return None
+            for runtime in layer.compatible_runtimes or []:
+                layer_folder = LAYER_SUBFOLDERS.get(runtime)
+
+                # unsupported runtime for layers
+                if not layer_folder:
+                    LOG.debug("Skipping symlink check for layer %s, unsupported runtime (%s)", layer.layer_id, runtime)
+                    continue
+
+                # not locally built layer
+                if not layer.codeuri:
+                    LOG.debug(
+                        "Skipping symlink check for layer %s, layer does not have locally defined resources",
+                        layer.layer_id,
+                    )
+                    continue
+
+                # eg. `.aws-sam/build/MyLayer` + `nodejs`
+                artifact_layer_path = Path(layer.codeuri, layer_folder)
+                # eg. `/opt` + `nodejs`
+                container_bind_path = Path(LambdaImage._LAYERS_DIR, layer_folder)
```

Comment:
Curious, what happens if there are multiple layers used by the function? Are we going to accept that last one will replace the earlier mappings?

Reply:
Updated, turns out that the tar library includes an option that lets it resolve symlinks for us. No need to write custom logic to do this.